### PR TITLE
Fix client secret MSA auth flow, other bug fixes

### LINKED ISSUE #33: Trying to access empty Shared With Me throws
If there is nothing Shared With Me and I try to access the Shared With Me 'folder', I get an exception saying the item wasn't found.  It would be more convenient if this function returned an empty Enumeration instead of throwing.

SDK version = 1.1.15
C# console app on Win 10-64

To reproduce:
```c#
// Use a 'home consumer' OneDrive account that has nothing shared with it.
OD.IDriveSharedWithMeRequestBuilder swmRequestBuilder = OneDrive.client.Drive.SharedWithMe();
OD.IDriveSharedWithMeCollectionPage result = swmRequestBuilder.Request().GetAsync().Result; // throws
```

### PR DESCRIPTION
**Pull request by ginach**
Fixes the fact we weren't piping client secret through correctly for MSA auth and adds tests, fixes #33

### CODE REVIEW

**Comment by aclev** [comment]
:+1:

File: src/OneDriveSdk/Requests/BaseClient.cs [hunk 1]
```diff
@@ -71,7 +71,7 @@ public IAuthenticationProvider AuthenticationProvider
         public string BaseUrl
         {
             get { return this.baseUrl; }
-            set { this.baseUrl = value.TrimEnd('/'); }
+            set { this.baseUrl = string.IsNullOrEmpty(value) ? null : value.TrimEnd('/'); }
```
Comment (peternied) [inline src/OneDriveSdk/Requests/BaseClient.cs]: Is this for ReSharper?  If you assign null, empty, whitespace to BaseUrl I would expect an InvalidArgument exception

File: src/OneDriveSdk/Requests/Extensions/OneDriveClientExtensions.cs [hunk 2]
```diff
@@ -268,11 +366,12 @@ public partial class OneDriveClient : IDisposable
         /// <param name="returnUrl">The application return URL for Microsoft account authentication.</param>
         /// <param name="scopes">The requested scopes for Microsoft account authentication.</param>
         /// <param name="clientSecret">The client secret for Microsoft account authentication.</param>
+        /// <param name="refreshToken">The refresh token for silent authentication.</param>
         /// <param name="serviceInfoProvider">The <see cref="IServiceInfoProvider"/> for initializing the <see cref="IServiceInfo"/> for the session.</param>
         /// <param name="credentialCache">The cache instance for storing user credentials.</param>
         /// <param name="httpProvider">The <see cref="IHttpProvider"/> for sending HTTP requests.</param>
         /// <returns>The <see cref="IOneDriveClient"/> for the session.</returns>
-        public static async Task<IOneDriveClient> GetSlientlyAuthenticatedMicrosoftAccountClient(
+        public static async Task<IOneDriveClient> GetSilentlyAuthenticatedMicrosoftAccountClient(
```
Comment (peternied) [inline src/OneDriveSdk/Requests/Extensions/OneDriveClientExtensions.cs]: Did you get a chance to spell check all the other code's functions?

**Comment by peternied** [comment]
:+1:

File: src/OneDriveSdk/Requests/Extensions/OneDriveClientExtensions.cs (hunk 2 above)
Comment (ginach) [inline src/OneDriveSdk/Requests/Extensions/OneDriveClientExtensions.cs]: I found a tool to spell check whole words but not method or class names. Including those fixes, though.

**Pull request merged**